src/utils/portals-common/portals-common.js

```diff
@@ -21,71 +24,152 @@ const getY = point => point[1];
 export const isPointInGeoJSONPolygon = isPointInPolygonWn(getX, getY);
 
 
+export const utcDateOnly = d => {
+  // Return a new Date object which is suitable for comparing UTC date only
+  // (fixed zero time component). One might suppose that the comparisons in
+  // the filter below also would work with local time (Date.setHours()), but
+  // they don't; only UTC works.
+  if (!d) {
+    return d;
+  }
+  const r = new Date(d);
+  r.setUTCHours(0, 0, 0, 0);
+  return r;
+};
+
+
+export const isNilOrAscending = (a, b, c) => {
+  // Allowing for nils (undefined, null), check that a <= b <= c.
+  return isNil(b) || ((isNil(a) || a <= b) && (isNil(c) || b <= c));
+};
+
+
+export const historyDateMatch = (
+  history, startDate, endDate, strict = true
+)  => {
+  // Return true if history observation dates match start and end date.
+  // Compare using dates only.
+  const minObsDate = utcDateOnly(history.min_obs_time);
+  const maxObsDate = utcDateOnly(history.max_obs_time);
+  startDate = utcDateOnly(startDate);
+  endDate = utcDateOnly(endDate);
+
+  if (strict) {
+    // Allowing for nils, check that
+    // minObsDate <= startDate <= endDate <= maxObsDate.
+    return (
+      isNilOrAscending(startDate, endDate)
+      && isNilOrAscending(minObsDate, startDate, maxObsDate)
+      && isNilOrAscending(minObsDate, endDate, maxObsDate)
+    );
+  }
+
+  // This is a variant of the looser legacy comparison. Allows nil start, end.
```

Comment:
Traditionally, a NULL endDate was used in the database to signify that a station was still *actively* recording data (the CRMP partners always wanted to differentiate between *active* stations vs. *historic* stations). Theoretically, there shouldn't be NULL startDates, but as you have found, there is no maintenance schedule for startDate and endDate (so they exist). That'd be worth a schema change (making `meta_history.sdate` not nullable) to correct.

Reply:
That schema change has been in the PyCDS issues for ... um ... a while now.